src/vcModals.cpp

```diff
@@ -325,24 +338,42 @@ void vcModals_DrawTileServer(vcState *pProgramState)
     static bool s_isDirty = false;
     static int s_currentItem = -1;
 
-    const char *pItems[] = { "png", "jpg" };
+    const char *pSlippyItems[] = { "png", "jpg" };
+    const char *pGoogleItems[] = { "Plan","Image","Terrain" };
     if (s_currentItem == -1)
     {
-      for (int i = 0; i < (int)udLengthOf(pItems); ++i)
-      {
-        if (udStrEquali(pItems[i], pProgramState->settings.maptiles.tileServerExtension))
-          s_currentItem = i;
-      }
+        if (pProgramState->settings.maptiles.serverFlag == vcMTSF_Slippy)
+        {
+            for (int i = 0; i < (int)udLengthOf(pSlippyItems); ++i)
+            {
+              if (udStrEquali(pSlippyItems[i], pProgramState->settings.maptiles.tileServerExtension))
+              {
+                s_currentItem = i;
+                break;
+              }
+            }
+        }
+        else if (pProgramState->settings.maptiles.serverFlag == vcMTSF_Google)
+        {
+            s_currentItem = pProgramState->settings.maptiles.mapTileType;
```

Comment:
No. `mapTileType` is an index into the `pGoogleItems` array that is saved to disk. This means that we can never modify the `pGoogleItems` array, or if we do, we have to create a whole bunch of nonsense to handle this lazy option. Mimic what the Slippy code does for the file extension, it's the same concept. Don't be lazy.